docs/corruption-detection-and-recovery.md

```diff
@@ -0,0 +1,62 @@
+# etcd Corruption Detection and Recovery
+
+Currently, in etcd 3.5.0-3.5.2, there is a low probability risk of data corruption across the etcd members [Upstream Issue](https://github.com/etcd-io/etcd/issues/13766), [Openshift Documentation](https://access.redhat.com/solutions/6849521). There is a mitigation coming to the Openshift Cluster etcd Operator that enables the etcd 'experimental-initial-corrupt-check' which should help detect corruption and stop the bad member from joining the cluster.
+This document will outline some ways to verify if an etcd member is corrupted so that you can be more confident that the corruption recovery steps are necessary and should be taken.
+
+# Corruption Diagnosis
+Unfortunately, there's no single command that will give a precise 'corrupt/not corrupt' at this time; however, if most or all of the following return that there might be corruption, then it's likely that there is corruption and recovery steps should be taken. Please note, that a single failure in the following commands does not necessarily mean that corruption has occurred.
+
+## Check Prometheus for etcd revision divergence
+Revision divergence is a natural happening of etcd since it takes time to sync a revision to all members; however, continuing divergence is an indication of corruption. If you have access to run Prometheus queries, you can check the following to graph the current revisions to visualize any divergence.
```

Comment:
What is the threshold of delta for deciding good or bad?  Trevor's alert may provide insight.

Reply:
In my testing, `etcd_debugging_mvcc_current_revision` often had running divergence and rarely zeroed in active clusters.  I'm using `etcd_debugging_mvcc_compact_revision` in #774 to avoid that, at the price of zero to five minutes of additional latency depending on where the hangup occurs within the 5m compaction period.